src/httpClient.js

```diff
@@ -172,16 +172,16 @@ mainHttpClientV2.interceptors.request.use(requestLargeDataOnFulfill, requestLarg
 mainHttpClient.interceptors.response.use(responseFulfillInterceptor, responseRejectInterceptor)
 mainHttpClientV2.interceptors.response.use(responseFulfillInterceptor, responseRejectInterceptor)
 
-export const showLargeResponsePopUp = setRequestErrorMessage => {
+export const showLargeResponsePopUp = (setRequestErrorMessage, customErrorMessage) => {
   if (!largeResponsePopUpIsOpen) {
     const errorMessage =
       'The query result is too large to display. Add a filter (or narrow it) to retrieve fewer results.'
 
-    setRequestErrorMessage(errorMessage)
+    setRequestErrorMessage(customErrorMessage || errorMessage)
     largeResponsePopUpIsOpen = true
 
     openPopUp(ConfirmDialog, {
-      message: errorMessage,
+      message: customErrorMessage || errorMessage,
```

Comment:
No need to have the condition 2 times. Define the message on the variable creation